presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayRemoveFunction.java

```diff
@@ -0,0 +1,132 @@
+import static com.facebook.presto.type.TypeUtils.parameterizedTypeName;
+import static com.facebook.presto.type.TypeUtils.readStructuralBlock;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+
+public final class ArrayRemoveFunction
+        extends ParametricScalar
+{
+    public static final ArrayRemoveFunction ARRAY_REMOVE_FUNCTION = new ArrayRemoveFunction();
+    private static final String FUNCTION_NAME = "array_remove";
+    private static final Signature SIGNATURE = new Signature(FUNCTION_NAME, ImmutableList.of(comparableTypeParameter("E")), "array<E>", ImmutableList.of("array<E>", "E"), false, false);
+
+    @Override
+    public Signature getSignature()
+    {
+        return SIGNATURE;
+    }
+
+    @Override
+    public boolean isHidden()
+    {
+        return false;
+    }
+
+    @Override
+    public boolean isDeterministic()
+    {
+        return true;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Remove specified values from the given array";
+    }
+
+    @Override
+    public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        checkArgument(types.size() == 1, format("%s expects only one argument", FUNCTION_NAME));
+        Type type = types.get("E");
+        TypeSignature valueType = type.getTypeSignature();
+        TypeSignature arrayType = parameterizedTypeName(StandardTypes.ARRAY, valueType);
+        MethodHandle methodHandle = methodHandle(ArrayRemoveFunction.class, "remove", Type.class, Slice.class, type.getJavaType());
+        Signature signature = new Signature(FUNCTION_NAME, arrayType, arrayType, valueType);
+        return new FunctionInfo(signature, getDescription(), isHidden(), methodHandle.bindTo(type), isDeterministic(), true, ImmutableList.of(false, false));
+    }
+
+    public static Slice remove(Type type, Slice array, Slice value)
+    {
+        return remove(type, array, createBlock(type, value));
+    }
+
+    public static Slice remove(Type type, Slice array, long value)
+    {
+        return remove(type, array, createBlock(type, value));
+    }
+
+    public static Slice remove(Type type, Slice array, double value)
+    {
+        return remove(type, array, createBlock(type, value));
+    }
+
+    public static Slice remove(Type type, Slice array, boolean value)
+    {
+        return remove(type, array, createBlock(type, value));
+    }
+
+    private static Slice remove(Type type, Slice array, Block valueBlock)
+    {
+        Block elementsBlock = readStructuralBlock(array);
+
+        int sizeAfterRemove = 0;
+        List<Integer> positions = new ArrayList<>();
+        for (int i = 0; i < elementsBlock.getPositionCount(); i++) {
+            if (!type.equalTo(elementsBlock, i, valueBlock, 0)) {
```

Comment:
Instead of using `type.equalTo` this should use a method handle, like `ArrayEqualOperator` and `ArrayJoinFunction` do